File: bindings/java/tests/identify.java
//
// identify.java
// 
// Java test for WS-Identify
//

import org.openwsman.Client;
import org.openwsman.ClientOptions;
import org.openwsman.OpenWSManConstants;
import org.openwsman.XmlDoc;

public class identify {

	public static void main(String argv[]) {
		Client c = new Client("http://wsman:secret@localhost:5985/wsman");
		ClientOptions op = new ClientOptions();
		op.set_dump_request();
		c.transport().set_auth_method(OpenWSManConstants.BASIC_AUTH_STR);

		XmlDoc doc = c.identify(op);
		if (doc == null || doc.isFault()) {
			System.err.println("Failed to identify: " + ((doc != null) ? doc.fault().reason() : "?"));
			System.exit(1);
		}

		System.out.println("ProtocolVersion: "
				+ doc.root().find(OpenWSManConstants.XML_NS_WSMAN_ID,
						OpenWSManConstants.WSMID_PROTOCOL_VERSION, 1));
		System.out.println("ProductVendor: "
				+ doc.root().find(OpenWSManConstants.XML_NS_WSMAN_ID,
						OpenWSManConstants.WSMID_PRODUCT_VENDOR, 1));
		System.out.println("ProductVersion: "
				+ doc.root().find(OpenWSManConstants.XML_NS_WSMAN_ID,
						OpenWSManConstants.WSMID_PRODUCT_VERSION, 1));
	}
}

File: bindings/ruby/tests/identify.rb
# test identify action

require 'test/unit'
require File.expand_path(File.join(File.dirname(__FILE__),'_loadpath'))
require 'openwsman'
require 'auth-callback'
require '_client'

class IdentifyTest < Test::Unit::TestCase
  def test_identify
    Openwsman::debug = 1
    
    client = Client.open
    assert client
    options = Openwsman::ClientOptions.new
    assert options
    options.set_dump_request

    doc = client.identify( options )
    unless doc
      puts "identify failed with #{client.last_error}:#{client.fault_string}"
      raise
    end
    assert doc
    puts "Error!" if doc.fault?
    root = doc.root
    assert root
    
#    print root
#
# root.find is the 'clumsy' way
#    prot_version = root.find( Openwsman::XML_NS_WSMAN_ID, "ProtocolVersion" )
#    prod_vendor = root.find( Openwsman::XML_NS_WSMAN_ID, "ProductVendor" )
#    prod_version = root.find( Openwsman::XML_NS_WSMAN_ID, "ProductVersion" )
    
    # Ruby allows to catch 'method_missing' making it much more elegant:
    prot_version = root.ProtocolVersion
    prod_vendor = root.ProductVendor
    prod_version = root.ProductVersion

    puts "\tProtocol #{prot_version}\n\tVendor #{prod_vendor}\n\tVersion #{prod_version}"
    
  end
end

